### ISSUE TITLE
Clarification on `carry` variable used during training

### ISSUE THREAD

**Issue by jren03** [danijar/dreamerv3#123]
Would you mind explaining the purpose of the `carry` variable used during training? From what I understand, it's a tuple of latent state and latent action.

From this [section](https://github.com/danijar/dreamerv3/blob/b1f27e7d6729705ec26c358aa74ead4fb9c1481b/dreamerv3/agent.py#L182C1-L187C32), it seems like `prevlat` is replaced with the context from the first sample of each batch. However, why does `prevact` or `carry[1]` not replaced in this case? Asked differently, if I were to alternate sampling from two different replay buffers during training, would I need need two different `carry` variables, or does a shared one suffice since it is getting replaced by context anyway?

Thank you in advance for the help!

**Comment by danijar** [comment]
You're right it's not needed with the default settings, but it allows truncated backprop through time by training on consecutive batches (https://github.com/danijar/dreamerv3/issues/157).
